assets/src/dashboard/app/api/api-provider.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { createContext, useCallback, useMemo } from 'react';
+import moment from 'moment';
+import queryString from 'query-string';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../config';
+import { STORY_STATUSES } from '../../constants';
+
+export const ApiContext = createContext({ state: {}, actions: {} });
+
+export function reshapeStoryObject({ id, title, modified, status }) {
+  return {
+    id,
+    status,
+    title: title.rendered,
+    modified: moment(modified),
+  };
+}
+
+export default function ApiProvider({ children }) {
+  const { api } = useConfig();
+
+  const fetchStories = useCallback(
+    async ({ status = STORY_STATUSES[0].value }) => {
+      try {
+        const path = queryString.stringifyUrl({
+          url: api.stories,
+          query: { status },
+        });
+        const serverStoryResponse = await apiFetch({
+          path,
+        });
+        return serverStoryResponse.map(reshapeStoryObject);
+      } catch (err) {
+        return [];
+      }
+    },
+    [api.stories]
+  );
+
+  const value = useMemo(
+    () => ({
+      state: {},
```

Comment:
Super nit, if state isn't being used, does it need to be included? 🤔

Reply:
Oh! I should probably have stories in state here.